pkg/kvstore/store/cell.go

```diff
@@ -19,25 +21,25 @@ var Cell = cell.Module(
 )
 
 type Factory interface {
-	NewSyncStore(clusterName string, backend SyncStoreBackend, prefix string, opts ...WSSOpt) SyncStore
-	NewWatchStore(clusterName string, keyCreator KeyCreator, observer Observer, opts ...RWSOpt) WatchStore
-	NewWatchStoreManager(backend WatchStoreBackend, clusterName string) WatchStoreManager
+	NewSyncStore(logger *slog.Logger, clusterName string, backend SyncStoreBackend, prefix string, opts ...WSSOpt) SyncStore
+	NewWatchStore(logger *slog.Logger, clusterName string, keyCreator KeyCreator, observer Observer, opts ...RWSOpt) WatchStore
+	NewWatchStoreManager(logger *slog.Logger, backend WatchStoreBackend, clusterName string) WatchStoreManager
```

Comment:
How about initializing the logger via hive (as part of `NewFactory`), rather than passing it explicitly here? That would allow to drop a few changes, and be consistent with the metrics initilization.